doc/users_guide.rst

```diff
@@ -0,0 +1,306 @@
+editable mode (including those that are not developped in that repository, but
+still included for convenience).
+
+Virtual machine installation
+++++++++++++++++++++++++++++++++++
+
+LISA provides a Vagrant recipe which automates the generation of a
+VirtualBox based virtual machine pre-configured to run LISA. To generate and
+use such a virtual machine you need:
+
+- `VirtualBox <https://www.virtualbox.org/wiki/Downloads>`__
+- `Vagrant <https://www.vagrantup.com/downloads.html>`__
+
+Once these two components are available on your machine, issue these commands::
+
+  git clone https://github.com/ARM-software/lisa.git -b next
+  cd lisa
+  vagrant up
+
+This last command builds and executes the VM according to the description provided
+by the Vagrant file available in the root folder of the LISA source tree.
+
+Once the VM installation is complete, you can access that VM with::
+
+  vagrant ssh
+
+LISA shell
+==========
+
+You should now have all of the required dependencies installed. From now on, you
+can use the LISA shell, which provides a convenient set of commands for easy
+access to many LISA related functions (done automatically by Vagrant)::
+
+  source init_env
+
+Run ``lisa-help`` to see an overview of the provided LISA commands.
+
+Notebooks
+=========
+
+The LISA shell can simplify starting an Jupyter notebook server::
+
+  [LISAShell lisa] \> lisa-jupyter start
+
+  Starting Jupyter Notebooks...
+
+  Notebook server configuration:
+    URL        :  http://127.0.0.1:8888/?token=b34F8D0e457BDa570C4A6D7AF113CB45d9CcAF44Aa7Cf400
+    Folder     :  /data/work/lisa/ipynb
+    Logfile    :  /data/work/lisa/ipynb/server.log
+    PYTHONPATH :
+	  /data/work/lisa/modules_root/
+
+
+  Notebook server task: [4] 30177
+
+Note that the ``lisa-jupyter`` command allows you to specify interface and
+port in case you have several network interfaces on your host::
+
+  lisa-jupyter start [interface [port]]
+
+The URL of the main folder served by the server is printed on the screen.
+By default it is http://127.0.0.1:8888/. Once the server is started you can
+have a look at the provided tutorial notebooks are accessible by following this
+`link <http://127.0.0.1:8888/notebooks/examples/typical_experiment.ipynb>`__.
+This initial tutorial can be seen (but not executed) also on `github
+<https://github.com/ARM-software/lisa/blob/next/ipynb/examples/typical_experiment.ipynb>`__.
+
+
+Transitioning from LISA legacy
+==============================
+
+A big refactoring effort was started in mid 2018, which produced a lot of
+(much needed) changes. If you are used to using LISA before the refactoring came
+into place, this guide is for you.
+
+Project structure
++++++++++++++++++
+
+* ``lisa/libs/utils`` is now ``lisa/lisa/``. ``lisa/libs/wlgen`` has also been
+  moved to that location.
+* devlib, trappy and bart are now under ``lisa/external/``. Git subtrees are
+  now used instead of submodules.
```

Comment:
I guess we could add a pointer to the contribution guide once it explains how to work with those subtrees.

Reply:
Yup!